### PR TITLE
[SuperEditor] Enable Mac shortcuts for iOS (Resolves #1609)

### PR DESCRIPTION
**Pull request by angelosilvestre**
[SuperEditor] Enable Mac shortcuts for iOS. Resolves #1609

Mac shortcuts, like cmd + c, don't execute for iOS. The reason is that the `isPrimaryShortcutKeyPressed` getter only checks if the current platform is mac.

This PR changes `isPrimaryShortcutKeyPressed` to consider iOS and also enable other shortcuts that are running for mac only.

### CODE REVIEW

File: super_editor/lib/src/default_editor/document_hardware_keyboard/document_keyboard_actions.dart [hunk 1]
```diff
@@ -221,7 +221,9 @@ ExecutionInstruction deleteDownstreamCharacterWithCtrlDeleteOnMac({
   required SuperEditorContext editContext,
   required KeyEvent keyEvent,
 }) {
-  if (defaultTargetPlatform != TargetPlatform.macOS) {
+  final isMacOrIos = defaultTargetPlatform == TargetPlatform.macOS || defaultTargetPlatform == TargetPlatform.iOS;
```
Comment (matthew-carroll) [inline super_editor/lib/src/default_editor/document_hardware_keyboard/document_keyboard_actions.dart:224]: How about we create a synthesized property somewhere called `isApple`? Do we already have some kind of platform class that accumulates static things like that?

File: super_editor/test/super_editor/supereditor_copy_and_paste_test.dart [hunk 2]
```diff
@@ -10,7 +9,7 @@ import 'supereditor_test_tools.dart';
 
 void main() {
   group('SuperEditor copy and paste > ', () {
-    testWidgetsOnMac('pastes within a paragraph', (tester) async {
+    testWidgetsOnMacAndIos('pastes within a paragraph', (tester) async {
```
Comment (matthew-carroll) [inline super_editor/test/super_editor/supereditor_copy_and_paste_test.dart:12]: We can call this `testWidgetsOnApple`

File: super_editor/lib/src/default_editor/document_hardware_keyboard/document_keyboard_actions.dart (hunk 1 above)
Comment (angelosilvestre) [reply super_editor/lib/src/default_editor/document_hardware_keyboard/document_keyboard_actions.dart]: Updated.

File: super_editor/test/super_editor/supereditor_copy_and_paste_test.dart (hunk 2 above)
Comment (angelosilvestre) [reply super_editor/test/super_editor/supereditor_copy_and_paste_test.dart]: Updated.

File: super_editor/lib/src/default_editor/document_hardware_keyboard/document_keyboard_actions.dart [hunk 3]
```diff
@@ -79,13 +79,11 @@ ExecutionInstruction scrollOnCtrlOrCmdAndHomeKeyPress({
     return ExecutionInstruction.continueExecution;
   }
 
-  final isMacOrIos = defaultTargetPlatform == TargetPlatform.macOS || defaultTargetPlatform == TargetPlatform.iOS;
-
-  if (isMacOrIos && !HardwareKeyboard.instance.isMetaPressed) {
+  if (isApple && !HardwareKeyboard.instance.isMetaPressed) {
```
Comment (matthew-carroll) [inline super_editor/lib/src/default_editor/document_hardware_keyboard/document_keyboard_actions.dart:82]: I don't think we want this in the global namespace. In my previous comment I asked if we had an existing class for platform properties like this. Do we have a place for it or no?

File: super_editor/lib/src/infrastructure/keyboard.dart [hunk 4]
```diff
@@ -34,8 +33,10 @@ enum ExecutionInstruction {
 
 extension PrimaryShortcutKey on KeyEvent {
   bool get isPrimaryShortcutKeyPressed =>
-      (defaultTargetPlatform == TargetPlatform.macOS && HardwareKeyboard.instance.isMetaPressed) ||
-      (defaultTargetPlatform != TargetPlatform.macOS && HardwareKeyboard.instance.isControlPressed);
+      (const [TargetPlatform.macOS, TargetPlatform.iOS].contains(defaultTargetPlatform) &&
```
Comment (matthew-carroll) [inline super_editor/lib/src/infrastructure/keyboard.dart:36]: These usages should be using the `isApple` check, too.

File: super_editor/lib/src/default_editor/document_hardware_keyboard/document_keyboard_actions.dart (hunk 3 above)
Comment (angelosilvestre) [reply super_editor/lib/src/default_editor/document_hardware_keyboard/document_keyboard_actions.dart:82]: I don't think we have a class for that.

Comment (angelosilvestre) [reply super_editor/lib/src/default_editor/document_hardware_keyboard/document_keyboard_actions.dart:82]: Should we create a class for that?

**Review by matthew-carroll — APPROVED** [review]
LGTM

**Pull request merged**